src/Sparse/ZZRow.jl

```diff
@@ -485,14 +485,24 @@ function ==(a::SRow{ZZRingElem, ZZRingElem_Array}, b::SRow{ZZRingElem, ZZRingEle
   return true
 end
 
-function sparse_row(R::ZZRing, A::Vector{Tuple{Int64, Int64}}; sort::Bool = true)
+function sparse_row(R::ZZRing, A::Vector{Tuple{Int, Int}}; sort::Bool = true)
   if sort && length(A) > 1
     A = Base.sort(A, lt=(a,b)->isless(a[1], b[1]))
   end
   a = ZZRingElem_Array()
   sizehint!(a, length(A))
   l = Int[]
-  for (p, v) = A
+  # assume A is sorted
+  local c
```

Comment:
Is there a reason that the check is inlined here instead of calling the helper function? Same question for the one below around line 520

Reply:
Yes, to not traverse the array twice.